src/misc/AttachShadowPolyfill.ts

```diff
@@ -24,6 +24,21 @@ export async function attachShadow(element: HTMLElement, options: IShadowOptions
   return shadowRoot;
 }
 
+export async function attachWithoutShadow(element: HTMLElement, options: IShadowOptions & ShadowRootInit): Promise<HTMLElement> {
```

Comment:
It seems the only difference with `attachShadow` function is the parent being an `iframe` or a `div` ?

Then I guess we could just add it as an option/argument for the function (like... `hostElement` in `IShadowOptions`). 

Is there any other difference I'm not seeing ?

Second thing I would suggest to try is if we could auto detect which host element (iframe or div) should be used from the context of the page.

Does it need to be an explicit option on the component ?
That options would be only be used in Salesforce integration, and then always set to true in Salesforce integration. And false everywhere else.

Could we try to detect "this environment will not properly support iframe, so let's use a div automatically" ?
That would remove the need for an option, and for the users to know how to configure this correctly.

Another thing is that by using a div, it means any styling applied to snippets *might* affect the rest of the page if not scoped properly. 

As long as that's okay for Salesforce integration...

Reply:
> Another thing is that by using a div, it means any styling applied to snippets might affect the rest of the page if not scoped properly.

That's why I made it an option so that existing stuff wouldn't be affected at least.

> Could we try to detect "this environment will not properly support iframe, so let's use a div automatically" ?
That would remove the need for an option, and for the users to know how to configure this correctly.

I like the idea of that as well and I thought about it, but I would be worried that we miss the mark just a little for some corner case and then someone gets stuck with a `div` when they want an `iframe` for example with no way of forcing it? That would be the only reason why I wouldn't do it automatically like you describe.

However I am willing to change the code to be just 1 method instead of two however, that I have no problem with. I made it separated to try to affect as little as possible the existing code to be sure to not break anything.

Reply:
`attachShadow` comes from the fact that it used to be "shadow DOM" I think. The name is already a bit misleading because of this legacy, so I wouldn't be against using the same method.

I also don't see the automatic detecting as really necessary

Reply:
I think manually specifying an option is good. I think it would tricky to auto-detect,

I am worried about Olivier's point about styling though. At the same time, not sure what we can do ...

Also, what about XSS attacks? It's harder because ML decides the html, but maybe we can prevent this by [encoding ](https://cheatsheetseries.owasp.org/cheatsheets/Cross_Site_Scripting_Prevention_Cheat_Sheet.html#rule-1-html-encode-before-inserting-untrusted-data-into-html-element-content) the content? I think there might be a lib already being used in another part of the JSUI

Reply:
I think filtering elements and attributes may be a good idea, even though ML does it too. Redundant security sounds best.

Reply:
Also, I agree about using an option. Perhaps you could add an `encapsulate: boolean` option.